addons/html_builder/static/src/plugins/shape/shape_selector.js

```diff
@@ -61,4 +58,19 @@ export class ShapeSelector extends BaseOptionComponent {
             }
         }
     }
+
+    onClose() {
+        this.rootRef.el.parentElement
+            .querySelector(".hb-sliding-panel-header button[aria-label='close']")
+            .click();
+    }
+
+    onFocusIn(ev) {
+        // Dispatch event to preview shape on focus
+        ev.target.dispatchEvent(new Event("pointerenter"));
+    }
+
+    onFocusOut(ev) {
+        ev.target.dispatchEvent(new Event("pointerleave"));
+    }
```

Comment:
Avoid those hacks and use the appropriate functions.
+ investigate to see why previews are triggered when the slide panel scrolls